lib/msf/core/post/dns/resolve_host.rb

```diff
@@ -0,0 +1,48 @@
+# -*- coding: binary -*-
+
+module Msf
+  class Post
+    module DNS
+      ###
+      #
+      # This module resolves session DNS
+      #
+      ###
+      module ResolveHost
+        # Takes the host name and makes use of nsloopup to resolve the IP
+        #
+        # @param [String] host Node for the ruby `send` method
+        # @return [String, nil] ip The resolved IP
+        def resolve_host(host)
```

Comment:
Should we skip address lookup if the host is already a dotted ipv4/ipv6 address?

Reply:
Good idea, I could add a check at the top of that method to just return the host if it is already a dotted ipv4/ipv6 address 👍

Reply:
The lazy approach suggested in #17920 and used in `post/windows/gather/enum_domain` is:

https://github.com/rapid7/metasploit-framework/blob/1ef030df7604bf188f26485e066e7c76d3ab5faa/modules/post/windows/gather/enum_domain.rb#L38-L39

I'm not sure if this supports IPv6.